outputs/jira_test.go

```diff
@@ -0,0 +1,1074 @@
+					priorityListJson, _ := json.Marshal(test.priorityList)
+					_, _ = w.Write(priorityListJson)
+				}
+			})
+			mux.HandleFunc("/rest/api/2/issue", func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to open issue" {
+					w.WriteHeader(http.StatusNotFound)
+					_, _ = w.Write([]byte("Failed to open issue"))
+				} else {
+					issueJson, _ := json.Marshal(test.issue)
+					_, _ = w.Write(issueJson)
+				}
+			})
+			ts := httptest.NewServer(mux)
+
+			savedCreateClient := createClient
+			createClient = func(ctx *JiraAPI) (*jira.Client, error) {
+				if test.wantError == "Failed to create client" {
+					return nil, fmt.Errorf(test.wantError)
+				} else {
+					return jira.NewClient(ts.Client(), ts.URL)
+				}
+			}
+			defer func() { createClient = savedCreateClient }()
+
+			err := test.jiraApi.Send(test.content)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestJiraAPI_OpenIssue(t *testing.T) {
+	tests := []struct {
+		name      string
+		issue     *jira.Issue
+		wantError string
+	}{
+		{
+			name:  "Happy path",
+			issue: &jira.Issue{ID: "issue1"},
+		},
+		{
+			name:      "sad path",
+			wantError: "open issue error",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError != "" {
+					w.WriteHeader(http.StatusNotFound)
+					_, _ = w.Write([]byte(test.wantError))
+				} else {
+					issueJson, _ := json.Marshal(test.issue)
+					_, _ = w.Write(issueJson)
+					w.WriteHeader(http.StatusOK)
+				}
+			}))
+			defer ts.Close()
+
+			jiraApi := &JiraAPI{}
+			jiraClient, err := jira.NewClient(ts.Client(), ts.URL)
+			if err != nil {
+				t.Fatalf("can't create jiraClient %v", err)
+			}
+
+			issue, err := jiraApi.openIssue(jiraClient, test.issue)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				assert.Equal(t, test.issue, issue)
+			}
+		})
+	}
+}
+
+func TestJiraAPI_CreateMetaProject(t *testing.T) {
+	tests := []struct {
+		name               string
+		metaInfo           interface{}
+		wantMetaProjectKey string
+		wantError          string
+	}{
+		{
+			name: "happy path",
+			metaInfo: jira.CreateMetaInfo{
+				Projects: []*jira.MetaProject{
+					{Key: "test"},
+					{Key: "debug"},
+				},
+			},
+			wantMetaProjectKey: "debug",
+		},
+		{
+			name:      "sad path (jira return error)",
+			metaInfo:  "bad struct",
+			wantError: "failed to get create meta : json: cannot unmarshal string into Go value of type jira.CreateMetaInfo",
+		},
+		{
+			name: "sad path (project not found)",
+			metaInfo: jira.CreateMetaInfo{
+				Projects: []*jira.MetaProject{
+					{Key: "test"},
+					{Key: "debug"},
+				},
+			},
+			wantMetaProjectKey: "bad project",
```

Comment:
it's not bad just something that was not found :). Also project key can't contain spaces. (just for authenticity)

Reply:
Changed